cc_plugin_ncei/ncei_base.py

```diff
@@ -0,0 +1,689 @@
+                    test_name = 'flag_masks'
+                    mask_length = len(getattr(dataset.variables[name], 'flag_masks', None)) #Used in length check
+                elif 'int' in str(dataset.variables[name].dtype):
+                    mask_val_check = hasattr(dataset.variables[name], 'flag_values')
+                    test_name = 'flag_values'
+                    mask_length = len(getattr(dataset.variables[name], 'flag_values', None)) #Used in length check
+                else:
+                    mask_val_check = False
+                    msgs = ['flag_masks or flag_values are not present']
+                    mask_length = 0
+                results.append(Result(BaseCheck.HIGH, mask_val_check, (name,test_name), msgs))
+
+                #Check Flag Meaning
+                msgs = []
+                if hasattr(dataset.variables[name],'flag_meanings'):
+                    meanings_check = True
+                    meaning_length = len(getattr(dataset.variables[name],'flag_meanings',None).split(','))
+                else:
+                    meanings_check = False
+                    msgs = ['flag_meanings not present']
+                    meaning_length = 0
+                results.append(Result(BaseCheck.HIGH, meanings_check, (name, 'flag_meanings'), msgs))
+
+                #Check Flag Meaning and Flag mask/Values same length
+                msgs = []
+                if mask_length == meaning_length:
+                    length_check = True
+                    msgs = []
+                else:
+                    length_check = False
+                    msgs = ['the length of flag_mask/values does not match flag_meanings']
+                results.append(Result(BaseCheck.HIGH, length_check, (name, 'flag_attributes_lengths'), msgs))
+
+                #Check Comment
+                msgs = []
+                if hasattr(dataset.variables[name], 'comment'):
+                    comment_check = True
+                else: 
+                    msgs = ['comment is missing']
+                    comment_check = False
+                results.append(Result(BaseCheck.MEDIUM, comment_check, (name,'comment'), msgs))
+     
+                #Check Long Name
+                msgs = []
+                if hasattr(dataset.variables[name], 'long_name'):
+                    long_check = True
+                else: 
+                    msgs = ['long_name not present']
+                    long_check = False
+                results.append(Result(BaseCheck.MEDIUM, long_check, (name,'long_name'), msgs)) 
+            
+                #Check References
+                msgs = []
+                if hasattr(dataset.variables[name], 'references'):
+                    source_check = True
+                else: 
+                    msgs = ['references is missing']
+                    source_check = False
+                results.append(Result(BaseCheck.MEDIUM, source_check, (name,'references'), msgs))
```

Comment:
For any checks involving a variable, please include the variable name in the message output.